### PR TITLE
Tree scaling

### PR DESCRIPTION
**Pull request by albert-smit**
Tree now automatically scales to fit inside the pane

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRingSegment.java [hunk 1]
```diff
@@ -44,17 +44,21 @@
      *            the X coordinate of the center of the circle
      * @param centerY
      *            the Y coordinate of the center of the circle
+     * @param scale
+     *            the scaling factor
      */
     public SunburstRingSegment(final PhylogeneticTreeItem value, final int layer,
             final double degreeStart, final double degreeEnd,
-            final double centerX, final double centerY) {
+            final double centerX, final double centerY, final double scale) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRingSegment.java]: 7 arguments is a bit too many. You could combine i.e. `centerX` and `centerY`.

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstView.java [hunk 2]
```diff
@@ -101,8 +111,11 @@ private void update() {
         // remove the old elements
         getChildren().clear();
 
+        centerX = getWidth() / 2d;
+        centerY = getHeight() / 2d;
+
         // add a center unit
-        SunburstCenter center = new SunburstCenter(currentItem);
+        SunburstCenter center = new SunburstCenter(currentItem, scale);
         center.setOnMouseClicked((mouseEvent) -> {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstView.java]: Useless parentheses

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstView.java [hunk 3]
```diff
@@ -168,4 +181,24 @@ private void drawRingRecursive(final PhylogeneticTreeItem node,
...
+    private double calculateScale() {
+        int depth = currentItem.maxDepth();
+        double minSize = Math.min(layoutBounds.getWidth(),
+                layoutBounds.getHeight());
+
+        double maxRadius = AbstractSunburstNode.CENTER_RADIUS;
+        maxRadius += depth * AbstractSunburstNode.RING_WIDTH;
+
+        double scale = minSize / (maxRadius * 2);
+        if (scale > 1) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstView.java]: Why not `Math.max` ?

**Comment by rutgervandenberg** [comment]
Looks fine.

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRingSegment.java [hunk 4]
```diff
@@ -87,10 +91,13 @@ public SunburstRingSegment(final PhylogeneticTreeItem value, final int layer,
      *            the X coordinate of the center of the circle
      * @param centerY
      *            the Y coordinate of the center of the circle
+     * @param scale
+     *            the scaling factor
      * @return a semi-circle with the specified dimensions
      */
     private Shape createRing(final int layer, final double degreeStart,
-            final double degreeEnd, final double centerX, final double centerY) {
+            final double degreeEnd, final double centerX, final double centerY,
```
Comment (alangerak) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRingSegment.java]: If you refactor above function to use point2d for centerX and centerY, don't forget to use this here as well

**Comment by alangerak** [comment]
Looks good otherwise

**Comment by joswinter** [comment]
Looks good.

**Comment by jorenham** [comment]
:+1:

**Pull request merged**